### PR TITLE
Convert user service to typescript

### PR DESCRIPTION
**Pull request by samuelim01**
## Description

The user service was converted from javascript to typescript
* Add typing
* Fix edge cases brought up by typescript
* Improve readability
* Add eslint, prettier and GHA for user service

## Checklist

- [x] I have updated documentation
- [x] All tests passing

## Screenshots (if applicable)

### CODE REVIEW

File: services/user/src/utils/helper.ts
Comment (LimZiJia) [inline services/user/src/utils/helper.ts:1]: Nice response handler.

File: services/user/src/middleware/basic-access-control.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    // request auth header: `Authorization: Bearer + <access_token>`
+    const token = authHeader.split(' ')[1];
+
+    if (!process.env.JWT_SECRET) {
+        handleInternalError(res, 'JWT Secret not provided');
+        return;
+    }
+
```
Comment (LimZiJia) [inline services/user/src/middleware/basic-access-control.ts:20]: When I first read this I was wondering if we need to verify if ```authHeader.split(' ')[0] === "Bearer"```. Functionally its going to be the same, but logically a bit wrong.

File: services/user/src/types/express/index.d.ts
Comment (LimZiJia) [inline services/user/src/types/express/index.d.ts:1]: Wow this is some magic

**Review by LimZiJia — APPROVED** [review]
Nice. Typing those Types must be frustrating. Good Job!

File: services/user/src/utils/helper.ts
Comment (samuelim01) [reply services/user/src/utils/helper.ts:1]: This was done by @KhoonSun47 in question service I'm just building on what he has already done 😆
